### PR TITLE
Add core messages

### PR DESCRIPTION
**Pull request by stevenheidel**
Add core messages.

This also required a change to the way the `*messages*` is generated, from a list to a hash-table with multiple namespaces.

### CODE REVIEW

File: update_python_spec.lisp [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
+  (rpcq::python-message-spec f (gethash :messages rpcq::*messages*))
+  (write-line "Wrote new messages.py"))
 
+; core-messages
+(rpcq::clear-messages :core-messages)
+(load "src/core-messages.lisp")
+(with-open-file (f "rpcq/core-messages.py"
+                   :direction ':output
+                   :if-exists ':supersede)
+  (rpcq::python-message-spec f (gethash :core-messages rpcq::*messages*))
```
Comment (stylewarning) [inline update_python_spec.lisp]:
```suggestion
  (rpcq::python-message-spec f (gethash ':core-messages rpcq::*messages*))
```

File: update_python_spec.lisp [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
+; messages
+(rpcq::clear-messages :messages)
 (load "src/messages.lisp")
 (with-open-file (f "rpcq/messages.py"
                    :direction ':output
-                   :if-exists ':supersede) 
-  (rpcq::python-message-spec f)
-  (write-line "Wrote new message spec."))
+                   :if-exists ':supersede)
+  (rpcq::python-message-spec f (gethash :messages rpcq::*messages*))
```
Comment (stylewarning) [inline update_python_spec.lisp]:
```suggestion
  (rpcq::python-message-spec f (gethash ':messages rpcq::*messages*))
```

File: update_python_spec.lisp [hunk 3]
```diff
@@ -1,11 +1,21 @@
...
+  (write-line "Wrote new messages.py"))
 
+; core-messages
+(rpcq::clear-messages :core-messages)
+(load "src/core-messages.lisp")
+(with-open-file (f "rpcq/core-messages.py"
+                   :direction ':output
```
Comment (stylewarning) [inline update_python_spec.lisp]: we should add
```
:if-does-not-exist ':create
```

File: update_python_spec.lisp [hunk 4]
```diff
@@ -1,11 +1,21 @@
...
+
+; messages
+(rpcq::clear-messages :messages)
 (load "src/messages.lisp")
 (with-open-file (f "rpcq/messages.py"
                    :direction ':output
-                   :if-exists ':supersede) 
-  (rpcq::python-message-spec f)
-  (write-line "Wrote new message spec."))
+                   :if-exists ':supersede)
```
Comment (stylewarning) [inline update_python_spec.lisp]: we should add
```
:if-does-not-exist ':create
```

File: update_python_spec.lisp (hunk 3 above)
Comment (stevenheidel) [reply update_python_spec.lisp]: So it already did that without me adding this?

**Comment by stevenheidel** [comment]
@stylewarning - I switched to filename instead of namespace and it cleans this up a lot, take a look at the latest diff

File: src/rpcq.lisp [hunk 5]
```diff
@@ -23,17 +23,12 @@
...
-
 ;; store all messages defined thus far in their namespace
-(defvar *messages* (make-hash-table))
+(defvar *messages* (make-hash-table :test 'equal))
 
-(defun clear-messages (namespace)
+(defun clear-messages ()
   "Clear the stored message definitions."
-  (setf (gethash namespace *messages*) nil))
+  (make-hash-table :test 'equal))
```
Comment (stylewarning) [inline src/rpcq.lisp]: if u want the most extreme efficiency u can make use of `clrhash` bringo gottem

File: update_python_spec.lisp [hunk 6]
```diff
@@ -1,21 +1,12 @@
...
-; core-messages
-(rpcq::clear-messages :core-messages)
-(load "src/core-messages.lisp")
-(with-open-file (f "rpcq/core-messages.py"
-                   :direction ':output
-                   :if-exists ':supersede)
-  (rpcq::python-message-spec f (gethash :core-messages rpcq::*messages*))
-  (write-line "Wrote new core-messages.py"))
+(dolist (namespace (list "messages" "core-messages"))
+  (load (format nil "src/~A.lisp" namespace))
```
Comment (stylewarning) [inline update_python_spec.lisp]: Welcome to the world where pathnames are objects. One can do:
```
(make-pathname :name namespace :type "lisp" :directory '(:RELATIVE "src"))
```

One can also do the slightly related:
```
(merge-pathnames (make-pathname :name namespace :type "lisp") "src/")
```
but it's not needed here.

File: update_python_spec.lisp [hunk 7]
```diff
@@ -1,21 +1,12 @@
...
+  (with-open-file (f (format nil "rpcq/~A.py" namespace)
```
Comment (stylewarning) [inline update_python_spec.lisp]: see `make-pathname` above.

File: update_python_spec.lisp [hunk 8]
```diff
@@ -1,21 +1,12 @@
...
+                     :direction ':output
+                     :if-exists ':supersede)
```
Comment (stylewarning) [inline update_python_spec.lisp]: `:if-does-not-exist ':create`

File: update_python_spec.lisp [hunk 9]
```diff
@@ -1,21 +1,12 @@
...
+    (rpcq::python-message-spec f (gethash namespace rpcq::*messages*))
+    (write-line (format nil "Wrote new ~A.py" namespace))))
```
Comment (stylewarning) [inline update_python_spec.lisp]: you can just `(format t "..." ...)` instead of `(write-line (format nil ...))`

**Comment by stylewarning** [comment]
@stevenheidel Can you resolve the Merge Conflict (TM)? Otherwise LGTM for merging

**Pull request merged**